isthmus/src/test/java/io/substrait/isthmus/ComparisonFunctionTest.java

```diff
@@ -0,0 +1,29 @@
+package io.substrait.isthmus;
+
+import java.io.*;
+import java.util.List;
+import org.junit.jupiter.api.Test;
+
+public class ComparisonFunctionTest extends PlanTestBase {
+
+  static List<String> CREATES =
+      List.of("CREATE TABLE numbers (c1 TINYINT, c2 TINYINT, c3 TINYINT)");
+
+  @Test
+  void between() throws Exception {
+    String query = "SELECT c1 FROM numbers WHERE c1 BETWEEN 1 AND 100";
+    assertSqlSubstraitRelRoundTrip(query, CREATES);
+  }
+
+  @Test
+  void coalesce() throws Exception {
+    String query = "SELECT coalesce(NULL, c1) FROM numbers";
+    assertSqlSubstraitRelRoundTrip(query, CREATES);
+  }
```

Comment:
This test doesn't exercise `coalesce` because Calcite converts it to a CASE expression.
```
LogicalProject(EXPR$0=[CASE(IS NOT NULL($0), $0, IS NOT NULL($1), $1, $2)])
  LogicalTableScan(table=[[NUMBERS]])
```